src/SmartComponents/modules/reducers.js

```diff
@@ -1,14 +1,51 @@
 import types from './types';
 
 const initialState = {
-    compare: {},
+    fullCompareData: {},
     addSystemModalOpened: false,
     selectedSystemIds: [],
     filterDropdownOpened: false,
     stateFilter: 'all',
-    factFilter: ''
+    factFilter: '',
+    totalFacts: 0,
+    page: 1,
+    perPage: 10,
+    filteredCompareData: {}
 };
 
+function paginateFilteredData(data, selectedPage, factsPerPage) {
+    let paginatedFacts = [];
+
+    for (let i = 0; i < data.length; i++) {
+        if (Math.ceil((i + 1) / factsPerPage) === selectedPage) {
+            paginatedFacts.push(data[i]);
+        }
+    }
+
+    return paginatedFacts;
+}
+
+function filterCompareData(data, stateFilter, factFilter) {
+    if (data === undefined) {
+        return {};
```

Comment:
Note this is a hash, but normally this function returns an array. I think we could drop this if statement all together.